frontend/src/modals/delete-document/DeleteDocumentModal.tsx

```diff
@@ -54,36 +107,46 @@ export const DeleteDocumentModal = ({ documentName, documentId, onDocumentDelete
   };
 
   useEffect(() => {
-    if (!loading) {
-      if (error) {
-        setErrorMessage(error.message);
-      } else if (data?.deleteDocument) {
-        const { deleteDocument } = data;
-        if (deleteDocument.__typename === 'DeleteDocumentSuccessPayload') {
-          onDocumentDeleted();
-        } else if (deleteDocument.__typename === 'ErrorPayload') {
-          setErrorMessage(deleteDocument.message);
-        }
-      }
+    if (deleteDocumentModal === 'success') {
+      onDocumentDeleted();
     }
-  }, [loading, data, error, onDocumentDeleted]);
+  }, [deleteDocumentModal, onDocumentDeleted]);
 
-  let bannerContent = errorMessage ? <Banner data-testid="banner" content={errorMessage} /> : null;
   return (
-    <Modal title={`Permanently delete "${documentName}"`} onClose={onClose}>
-      <div className={styles.container}>
-        <div className={styles.content}>
-          <Text className={styles.subtitle}>
+    <>
+      <Dialog open={true} onClose={onClose} aria-labelledby="dialog-title" fullWidth>
+        <DialogTitle id="dialog-title">{`Permanently delete "${documentName}"`}</DialogTitle>
+        <DialogContent>
+          <DialogContentText>
             This action will permanently delete the document and all its associated content. These items will no longer
             be accessible to you or anyone else. This action is irreversible.
-          </Text>
-          <div className={styles.bannerArea}>{bannerContent}</div>
-        </div>
-        <Buttons>
-          <DangerButton type="button" onClick={onDeleteDocument} label="Delete" data-testid="delete-document" />
-        </Buttons>
-      </div>
-    </Modal>
+          </DialogContentText>
+        </DialogContent>
+        <DialogActions>
+          <Button variant="contained" onClick={onDeleteDocument} color="primary" data-testid="delete-document">
+            Delete
+          </Button>
```

Comment:
Could you make this button enabled only in the pristine state to prevent double clicking on it?

Reply:
Done